### PR TITLE
Issue 12815 cluster autowire

### PR DESCRIPTION
**Pull request by wezell**
This makes the Server object pretty much immutable, removes cluster_server_uptime, uses the sitelic as the heartbeat,

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java [hunk 1]
```diff
@@ -1,85 +1,318 @@
...
+	public final String serverId;
+	public final  String clusterId;
+	public final  String ipAddress;
+	public final  String host;
+	public final  String name;
+	public final  Integer esTransportTcpPort;
+	public final  Integer esNetworkPort;
+	public final  Integer esHttpPort;
+	public final  Integer cachePort;
+	public final  Date lastHeartBeat;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java]: needs to be private to insure immutability

File: dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java [hunk 2]
```diff
@@ -1,85 +1,318 @@
...
+			return this;
+		}
+
+		public Builder withCachePort(Integer cachePort) {
+			this.cachePort = cachePort;
+			return this;
+		}
+
+		public Builder withLastHeartBeat(Date lastHeartBeat) {
+			this.lastHeartBeat = lastHeartBeat;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java]: this.lastHeartBeat = new Date(lastHeartBeat.getTime()) to insure immutability

File: dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java [hunk 3]
```diff
@@ -1,85 +1,318 @@
...
+
 	public Integer getCachePort() {
 		return cachePort;
 	}
-	public void setCachePort(Integer cachePort) {
-		this.cachePort = cachePort;
-	}
+
 	public Date getLastHeartBeat() {
 		return lastHeartBeat;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java]: return defensive copy "new Date(lastHeartBeat.getTime())"  to make it immutable

File: dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java (hunk 1 above)
Comment (wezell) [reply dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java]: but they are final once the class is built

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/cluster/bean/Server.java]: applies for the Date, which is mutable. Nothing prevents doing this:
Date myDate = server.lastHeartBeat;
myDate.setTime(millis); 
which effectively modifies the state of "server"

**Review by dsilvam — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**